### PR TITLE
Performance Display Component

### LINKED ISSUE #127: Performance (Achievement) display component
## Description
As a User,
I need to get the sense of completing an activity/game or stage,
so that I can track my progress and get the sense of accomplishment.

### Draft mockup
achievement

### Update [Required]
- [ ] Developing the achievement page or modal.
- [ ] Adding transitions to it.
- [ ] Unit Testing coverage > 80%
- [ ] Pass Travis Build

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
10-12 hours

### PR DESCRIPTION
**Pull request by lunayach**
### Performance Display Component
Made the component that would appear after completion of each activity.

Fixes #127 

### Type of Change:

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.
A jasmine test for performance display component was written => it passed.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by lunayach** [comment]
I have currently integerated to one component only. Please approve so that I add to other componets as well.
display

**Comment by Raxerz** [comment]
The UI looks good now @lunayach . Can the options info kit, home, and next not have the square border?

File: src/app/performanc-dispaly.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    TestBed.configureTestingModule({
+      providers: [PerformanceDisplayService],
+      imports: [MatDialogModule]
+    });
+  });
+
+  it('should be created', inject([PerformanceDisplayService], (service: PerformanceDisplayService) => {
+    expect(service).toBeTruthy();
+  }));
+});
```
Comment (Raxerz) [inline src/app/performanc-dispaly.service.spec.ts]: new line

File: src/app/performance-display/models/SuccessMessage.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class ActivitySuccessObject {
+  currentActivityName: string;
+  nextActivityURL: string;
+  nextActivity: string;
+}
```
Comment (Raxerz) [inline src/app/performance-display/models/SuccessMessage.ts]: new line

File: src/app/performance-display/performance-display.component.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  font-size: $font-size-small;
+  font-weight: $bold;
+  color: $grey;
+}
+
+.row {
+  color: $blue;
+  display: flex;
+  justify-content: space-between;
+}
```
Comment (Raxerz) [inline src/app/performance-display/performance-display.component.scss]: new line

File: src/app/performance-display/performance-display.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should have injected values', () => {
+    expect(component.nextActivity).toEqual('nextActivity');
+    expect(component.nextActivityURL).toEqual('/test');
+    expect(component.activityName).toEqual('TestActivity');
+  });
+});
```
Comment (Raxerz) [inline src/app/performance-display/performance-display.component.spec.ts]: new line

File: src/app/services/performance-display.service.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  openDialog(Message: ActivitySuccessObject) {
+    const dialogConfig = new MatDialogConfig();
+
+    dialogConfig.disableClose = true;
+    dialogConfig.autoFocus = true;
+    dialogConfig.data = Message;
+    dialogConfig.width = '500px';
+
+
```
Comment (paavininanda) [inline src/app/services/performance-display.service.ts]: The excessive new lines can be removed from this file.

File: src/app/performance-display/models/SuccessMessage.ts (hunk 2 above)
Comment (lunayach) [reply src/app/performance-display/models/SuccessMessage.ts]: @Raxerz by newline do we mean a new empty line after the brace? If that's the thing, then I have done that.

Comment (lunayach) [reply src/app/performance-display/models/SuccessMessage.ts]: Done.

File: src/app/performance-display/performance-display.component.spec.ts (hunk 4 above)
Comment (lunayach) [reply src/app/performance-display/performance-display.component.spec.ts]: Done.

**Comment by shreyans29** [comment]
UI looks very good now.

File: src/app/performance-display/performance-display.component.scss [hunk 6]
```diff
@@ -0,0 +1,26 @@
+@import "../styles/main";
+
+#title {
+  text-align: center;
+  color: $black;
```
Comment (shreyans29) [inline src/app/performance-display/performance-display.component.scss]: Exactly, this is the right way, nice. But you still have to change it in the Strides PR.

File: src/app/performance-display/performance-display.component.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+@Component({
+  selector: 'app-performance-display',
+  templateUrl: './performance-display.component.html',
+  styleUrls: ['./performance-display.component.scss']
+})
+export class PerformanceDisplayComponent implements OnInit {
+  public SuccessObject: ActivitySuccessObject;
+  public language: any;
+  public activityName: string;
```
Comment (shreyans29) [inline src/app/performance-display/performance-display.component.ts]: can this be a model ?

Comment (lunayach) [reply src/app/performance-display/performance-display.component.ts]: @shreyans29 Yes! ActivitySuccessObject I created for that purpose only. These activityName and other variables are for binding in the template.

**Comment by lunayach** [comment]
@shreyans29 Okay! so like displaying the performance display component only after completing for the first time. Right?

**Comment by lunayach** [comment]
@Raxerz fixed that border in the buttons, issue.

**Comment by Raxerz** [comment]
@lunayach change the cursor to pointer when hovered over the 3 options (info-kit, dashboard and peace corps policy)

**Comment by lunayach** [comment]
@Raxerz @shreyans29 @paavininanda To integrate with all activities, I am thinking of making a separate config file having an array of all the current activities. So, in future, if there is a change required in order of activities, we would need to change their order in the array and we are good to go.
Shall I go ahead?

**Comment by Raxerz** [comment]
@lunayach are you saying that you'll have an array of activity headers and then use the same template to just change the text in the dialog, with the dialog will be one single implementation?

**Comment by paavininanda** [comment]
@lunayach I also think it is a good idea 👍

**Comment by Raxerz** [comment]
Yeah that's a good idea @lunayach. Please go ahead.

**Comment by lunayach** [comment]
@Raxerz done!

**Pull request merged**